### PR TITLE
Introduce error_chain

### LINKED ISSUE #183: Use error-chain for error handling

### PR DESCRIPTION
**Pull request by tomusdrw**
Closes #183

### CODE REVIEW

File: http/src/error.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+use hyper;
+
+error_chain! {
+	foreign_links {
+		Io(io::Error);
+	}
+
+	errors {
+		/// Hyper error
+		Hyper(e: hyper::Error) {
```
Comment (debris) [inline http/src/error.rs]: why not `foreign_links`? It [implements](https://docs.rs/hyper/0.11.6/hyper/error/enum.Error.html#impl-StdError) `std::error::Error`

File: ws/src/error.rs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+	errors {
+		/// Attempted action on closed connection.
+		ConnectionClosed {
+			description("connection is closed"),
+			display("Action on closed connection."),
+		}
+
+		/// WebSockets error.
+		WebSocket(t: ws::Error) {
```
Comment (debris) [inline ws/src/error.rs]: why not `foreign_links`?

File: http/src/error.rs (hunk 1 above)
Comment (tomusdrw) [reply http/src/error.rs]: Explanation given in custom `From` implementation. We want to unwrap `Io` error from hyper error.

File: ws/src/error.rs (hunk 2 above)
Comment (tomusdrw) [reply ws/src/error.rs]: Same as above.

File: http/src/error.rs (hunk 1 above)
Comment (debris) [reply http/src/error.rs]: but why?

**Pull request merged**